### PR TITLE
Hardened type resolution checks in QueryParameterizer

### PR DESCRIPTION
**Pull request by andrecsilva**
Also updated the README to mention docker for running tests.

### CODE REVIEW

**Comment by nahsra** [comment]
Does this fix a bug? Do we have a regression test?

**Comment by andrecsilva** [comment]
> Does this fix a bug? Do we have a regression test?

The SQLParameterizer codemod was crashing on this file:

https://github.com/HCL-TECH-SOFTWARE/AltoroJ/blob/1a47e993a2db5c5c4fc42b61feab03704b9ae29c/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L212

This PR fixes that.

The core issue is with JavaParser's type resolution for some expressions. Sometimes JavaParser's type resolution will fail (and throw an exception). It happened before and I'm not entirely sure how to reproduce it, hence no tests.

It is likely this also affects other codemods that uses type resolution. I could investigate and write some tests for this issue if you feel it's worth pursuing it.

**Comment by nahsra** [comment]
I've had to make similar changes in that file as you can probably see.

The code appears to be [Apache license](https://github.com/HCL-TECH-SOFTWARE/AltoroJ/blob/1a47e993a2db5c5c4fc42b61feab03704b9ae29c/LICENSE), so I suppose we could bring in a minimal version of the repo for a test case. I'll leave that up to you.

File: core-codemods/src/main/java/io/codemodder/codemods/QueryParameterizer.java [hunk 1]
```diff
@@ -152,14 +171,17 @@ private boolean convertibleToString(final Expression exp) {
     // See
     // https://download.oracle.com/otn-pub/jcp/jdbc-4_2-mrel2-spec/jdbc4.2-fr-spec.pdf
     // for type conversions by setObject
-    final var type = exp.calculateResolvedType();
```
Comment (nahsra) [inline core-codemods/src/main/java/io/codemodder/codemods/QueryParameterizer.java:155]: I would do something like 

```
typeRef = calculateResolvedType(exp);
if (typeRef.isEmpty()) {
  return false;
}
var type = typeRef.get();
// now just use `type` everywhere
```

It's a cleaner diff for this PR, but it also just makes the code easier to read so we don't have `get()` scattered everywhere.

**Review by nahsra — CHANGES_REQUESTED** [review]

File: core-codemods/src/main/java/io/codemodder/codemods/QueryParameterizer.java [hunk 2]
```diff
@@ -54,7 +54,7 @@ List<Deque<Expression>> getInjections() {
     return injections;
   }
 
-  private Optional<ResolvedType> calculateResolvedType(Expression e) {
+  private final Optional<ResolvedType> calculateResolvedType(Expression e) {
```
Comment (nahsra) [inline core-codemods/src/main/java/io/codemodder/codemods/QueryParameterizer.java:57]: I meant `final` on the `Expression e` 😄

Comment (andrecsilva) [reply core-codemods/src/main/java/io/codemodder/codemods/QueryParameterizer.java:56]: Afters months of looking mostly at python code, It seems I can barely remember what `final` means.

**Review by nahsra — APPROVED** [review]

**Pull request merged**